File: app/src/main/java/com/emiryanvl/tapreader/data/remote/dataSources/RemoteBookDataSource.java
package com.emiryanvl.tapreader.data.remote.dataSources;

import com.emiryanvl.tapreader.domain.model.Book;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

public interface RemoteBookDataSource {

    Observable<List<Book>> getBooksByQuery(String query);

    Observable<List<Book>> getBooksBySubject(String subject);
}

File: app/src/main/java/com/emiryanvl/tapreader/data/remote/dataSources/RemoteBookDataSource.kt
package com.emiryanvl.tapreader.data.remote.dataSources

import com.emiryanvl.tapreader.domain.models.Book
import kotlinx.coroutines.flow.Flow

interface RemoteBookDataSource {

    fun getBooksByQuery(query: String): Flow<List<Book>>

    fun getBooksBySubject(subject: String): Flow<List<Book>>
}